### PR TITLE
Add step configuration to specify match order

### LINKED ISSUE #18: Add ability to allow for out of order processing
Similar to the ability that allows for substring matches, the ability for the expected_stdout_lines and expected_stderr_lines to be considered as a Set and allowing out of order processing would be helpful for validating some of the examples. 

adding a field like: 
```yaml
processing_order: inorder/outoforder
```
will be great. Or also adding it to the existing output_match_mode will also be great.

### PR DESCRIPTION
**Pull request by shubham1172**
This PR introduces a new step configuration: `match_order`.

If `match_order` is `sequential`, the expected lines are expected to be a part of output **in the same order**. This is the existing behavior and also the default (i.e. if `match_order` is missing).

If `match_order` is `none`, the expected lines can be **in any order** in the output.

Closes #18

### CODE REVIEW

**Review by mukundansundar — APPROVED** [review]
Lgtm

@berndverst thoughts?

**Comment by shubham1172** [comment]
FYI, I moved to Python 3.8 to use the walrus operator `:=`.

**Comment by berndverst** [comment]
Python 3.7 is still supported and we run Python 3.7 tests with mechanical markdown in the Python SDK. You must retain Python 3.7 support.

**Comment by shubham1172** [comment]
> Python 3.7 is still supported and we run Python 3.7 tests with mechanical markdown in the Python SDK. You must retain Python 3.7 support.

Thanks, reverted back.

File: mechanical_markdown/__init__.py [hunk 1]
```diff
@@ -7,6 +7,6 @@
 from mechanical_markdown.recipe import Recipe as MechanicalMarkdown
 from mechanical_markdown.parsers import MarkdownAnnotationError
 
-__version__ = '0.5.2'
+__version__ = '0.5.3'
```
Comment (berndverst) [inline mechanical_markdown/__init__.py:10]: Let's version this as 0.6.0. This is major enough!

```suggestion
__version__ = '0.6.0'
```

**Review by berndverst — CHANGES_REQUESTED** [review]
The `examples` folder markdown itself is an integration test that uses mechanical markdown.

Please add a section to `https://github.com/dapr/mechanical-markdown/tree/main/examples/README.md` which shows how to use the new option, and which actually has the mechanical markdown annotation.

For this example, please do the following:
- Use Match Order `None`
- Assert expected output lines to be:
```
- "first"
- "second"
```

But in your example code block do:
```bash
echo "second"
echo "first"
```

This integration test should pass and also function as an example!

**Comment by shubham1172** [comment]
Somehow the build action is not getting triggered, but I validated the example locally - 
`python3 -m mechanical_markdown ./examples/io.md`

image

**Review by berndverst — APPROVED** [review]
Thanks for adding the example and integration test! LGTM

**Comment by berndverst** [comment]
@shubham1172 i can see this test ran part of CI/CD. It's the "examples" step as part of the build workflow. The third run on this PR seems to include it.

**Comment by berndverst** [comment]
@artursouza please merge with STC power, unless @wcs1only wants to merge and happens to see this :)

**Review by artursouza — APPROVED** [review]

**Pull request merged**